pkg/engine/engine.go

```diff
@@ -1197,7 +1355,7 @@ func (e *Engine) notifierWorker(ctx context.Context) {
 			result.SourceType == sourcespb.SourceType_SOURCE_TYPE_POSTMAN) {
 			continue
 		}
-		e.dedupeCache.Add(key, result.DecoderType)
+		e.dedupeCache.Set(key, result.DecoderType)
```

Comment:
How does the cache handle indeterminate errors? 
- status code 429
- i/o timeout
- etc.